mbf_costmap_nav/src/mbf_costmap_nav/search_helper.cpp

```diff
@@ -0,0 +1,294 @@
+// mbf_costmap_nav
+#include "mbf_costmap_nav/search_helper.h"
+
+// std
+#include <queue>
+#include <unordered_set>
+
+namespace mbf_costmap_nav
+{
+
+SearchHelper::SearchHelper(const costmap_2d::Costmap2DROS* costmap, const SearchConfig& config,
```

Comment:
Can this class work on a reference instead? Why passing a RAW pointer all the time to all the functions? You never check for nullptr at any place either.

Reply:
changing to const ref